src/sentry/options/types.py

```diff
@@ -0,0 +1,173 @@
+"""
+sentry.options.types
+~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2015 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+from __future__ import absolute_import, print_function
+
+from sentry.utils.yaml import safe_load
+from yaml.parser import ParserError
+from yaml.scanner import ScannerError
+
+__all__ = (
+    'InvalidTypeError',
+    'Any', 'Bool', 'Int', 'Float', 'String', 'Dict', 'Sequence',
+)
+
+
+class InvalidTypeError(TypeError):
+    pass
+
+
+class OptionType(object):
+    """Base OptionType that provides type coersion"""
+    name = ''
+    # Default value to be returned when initializing
+    default = None
+    # Types that do not need to be coerced
+    expected_types = (object,)
+    # Types that are acceptable for coersion
+    compatible_types = (basestring,)
+
+    def __call__(self, value=None):
+        if value is None:
+            return self._default()
+        if self.test(value):
+            return value
+        if isinstance(value, self.compatible_types):
+            rv = self.convert(value)
+            # Make sure convert was able to do the right thing
+            # and give us the type we were expecting
+            if self.test(rv):
+                return rv
+        raise InvalidTypeError('{!r} is not a valid {}'.format(value, repr(self)))
+
+    def convert(self, value):
+        return value
+
+    def _default(self):
+        return self.default
+
+    def test(self, value):
+        """Check if the value needs to be coerced or not"""
+        return isinstance(value, self.expected_types)
+
+    def __repr__(self):
+        return self.name
+
+
+class AnyType(OptionType):
+    """A type that accepts any value and does no coersion"""
+    name = 'any'
+
+
+class BoolType(OptionType):
+    "Coerce a boolean from a string"
+    name = 'boolean'
+    default = False
+    expected_types = (bool,)
+
+    def convert(self, value):
+        if value.lower() in ('y', 'yes', 't', 'true', '1'):
+            return True
+        if value.lower() in ('n', 'no', 'f', 'false', '0'):
+            return False
```

Comment:
(Continuing the thread from above.) It's weird to me that this method can return a value that doesn't match the expected types but that will cause an exception in another part of the codebase rather than where validation occurs.